unity-game/Assets/Scripts/ElevatorScripts/AutoElevator.cs

```diff
@@ -0,0 +1,220 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AutoElevator : MonoBehaviour
+{
+    public bool notMoving;
+    public bool reachedDestination;
+    public bool onStartPoint;
+    public bool onEndPoint;
+
+    public GameObject objectToMove;
+    public GameObject theRightWall;
+    public GameObject theLeftWall;
+    public GameObject theStartLeftBarrier;
+    public GameObject theStartRightBarrier;
+    public GameObject theEndLeftBarrier;
+    public GameObject theEndRightBarrier;
+    public GameObject theStartPoint;
+    public GameObject theEndPoint;
+
+    public Transform startPoint;
+    public Transform endPoint;
```

Comment:
The elevator was originally based on a tutorial, so that's a remnant from that. As for whether the transform can indeed be gained from the GameObject, I'm not sure - if it's possible, I don't know how to do it.  :/